driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/gridfs/ResizingByteBufferFlux.java

```diff
@@ -75,14 +75,14 @@ protected void hookOnNext(@NonNull final ByteBuffer value) {
                     } else {
                         byte[] byteArray = new byte[remainder.remaining() + value.remaining()];
                         ByteBuffer newBuffer = ByteBuffer.wrap(byteArray);
-                        copyByteBuffer(remainder.asReadOnlyBuffer(), newBuffer);
```

Comment:
turning into read only buffers meant we couldn't use system array copies - so adding the test for this was a great catch.